src/legacy/core_plugins/kibana/public/management/sections/objects/lib/__jest__/import_legacy_file.test.js

```diff
@@ -20,44 +20,18 @@
 import { importLegacyFile } from '../import_legacy_file';
 
 describe('importFile', () => {
-  it('should import a file', async () => {
-    class FileReader {
-      readAsText(text) {
-        this.onload({
-          target: {
-            result: JSON.stringify({ text }),
-          },
-        });
-      }
-    }
+  it('should import a file with valid json format', async () => {
+    const file = new File([`{"text": "foo"}`], 'file.json');
 
-    const file = 'foo';
-
-    const imported = await importLegacyFile(file, FileReader);
-    expect(imported).toEqual({ text: file });
+    const imported = await importLegacyFile(file);
+    expect(imported).toEqual({ text: 'foo' });
```

Comment:
Was using a mocked FileReader implementation, therefor testing only a small part of the `importFile` function. Adapted the tests to use a real `File` as input